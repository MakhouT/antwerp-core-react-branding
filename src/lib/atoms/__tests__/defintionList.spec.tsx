import * as React from "react";

import { shallow } from "enzyme";

import * as A from "../..";
import { renderAndMatchSnapshot } from "../../../__tests__/__fixtures__/testUtils";

describe("Copyright", () => {
    it("Renders default", () => {
        // Arrange
        const list = {
            "abrac-adabra": "some word",
            "aya sofia": "some building",
            "rubicon": "some river",
        };
        const element = <A.DefinitionList list={list} />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("Renders capitalized", () => {
        // Arrange
        const list = {
            "abrac-adabra": "some word",
            "aya sofia": "some building",
            "rubicon": "some river",
        };
        const element = <A.DefinitionList list={list} capitalize />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("Renders capitalized and with colon", () => {
        // Arrange
        const list = {
            "abrac-adabra": "some word",
            "aya sofia": "some building",
            "rubicon": "some river",
        };
        const element = <A.DefinitionList list={list} capitalize colon />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("Renders null list without error", () => {
        // Arrange
        const element = <A.DefinitionList list={null} capitalize colon />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("Renders exact elements", () => {
        // Arrange
        const list = {
            "abrac-adabra": "some word",
            "aya sofia": "some building",
            "rubicon": "some river",
        };
        const expected = Object.keys(list).length;
        const element = <A.DefinitionList list={list} />;

        // Act
        const mounted = shallow(element);

        // Assert
        expect(mounted.find("dt").children().length).toBe(expected);
        expect(mounted.find("dd").children().length).toBe(expected);
    });

    it("Renders capitalized, check content", () => {
        // Arrange
        const list = {
            "abrac-adabra": "some word",
            "aya sofia": "some building",
            "rubicon": "some river",
        };
        const expected = "Abrac-adabra";
        const element = <A.DefinitionList list={list} capitalize />;

        // Act
        const mounted = shallow(element);

        // Assert
        expect(mounted.find("dt").first().text()).toBe(expected);
    });

    it("Renders capitalized, check content, edge-case", () => {
        // Arrange
        const list = {
            "a": "some word",
            "aya sofia": "some building",
            "rubicon": "some river",
        };
        const expected = "A";
        const element = <A.DefinitionList list={list} capitalize />;

        // Act
        const mounted = shallow(element);

        // Assert
        expect(mounted.find("dt").first().text()).toBe(expected);
    });

});
