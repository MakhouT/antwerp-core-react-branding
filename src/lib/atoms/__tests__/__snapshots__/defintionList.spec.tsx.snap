// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copyright Renders capitalized 1`] = `
<dl
  className="a-definition-list"
>
  <dt>
    Abrac-adabra
  </dt>
  <dd>
    some word
  </dd>
  <dt>
    Aya sofia
  </dt>
  <dd>
    some building
  </dd>
  <dt>
    Rubicon
  </dt>
  <dd>
    some river
  </dd>
</dl>
`;

exports[`Copyright Renders capitalized and with colon 1`] = `
<dl
  className="a-definition-list"
>
  <dt>
    Abrac-adabra:
  </dt>
  <dd>
    some word
  </dd>
  <dt>
    Aya sofia:
  </dt>
  <dd>
    some building
  </dd>
  <dt>
    Rubicon:
  </dt>
  <dd>
    some river
  </dd>
</dl>
`;

exports[`Copyright Renders default 1`] = `
<dl
  className="a-definition-list"
>
  <dt>
    abrac-adabra
  </dt>
  <dd>
    some word
  </dd>
  <dt>
    aya sofia
  </dt>
  <dd>
    some building
  </dd>
  <dt>
    rubicon
  </dt>
  <dd>
    some river
  </dd>
</dl>
`;

exports[`Copyright Renders null list without error 1`] = `
<dl
  className="a-definition-list"
/>
`;
